### PR TITLE
Upgrade to rustc 1.31.0-nightly (8c4ad4e9e 2018-10-04)

### PR DESCRIPTION
**Pull request by SimonSapin**
CC https://github.com/rust-lang/rust/issues/54846

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/21872)

### CODE REVIEW

File: components/script/dom/bindings/codegen/CodegenRust.py [hunk 1]
```diff
@@ -2278,7 +2278,10 @@ def define(self):
         return """\
 static INTERFACE_OBJECT_CLASS: NonCallbackInterfaceObjectClass =
     NonCallbackInterfaceObjectClass::new(
-        &%(constructorBehavior)s,
+        {
```
Comment (nox) [inline components/script/dom/bindings/codegen/CodegenRust.py]: Can you add an issue link?

**Comment by jdm** [comment]
@bors-servo retry
* #21873

**Comment by SimonSapin** [comment]
Amended to replace "now" with "nightly-2018-10-05" in the code comment.

@bors-servo r=nox

**Pull request merged**